Tests/Pcap++Test/Tests/FileTests.cpp

```diff
@@ -974,3 +974,44 @@ PTF_TEST_CASE(TestSolarisSnoopFileRead)
 
 	readerDev.close();
 }  // TestSolarisSnoopFileRead
+
+PTF_TEST_CASE(TestPcapFileWriterDeviceDestructor)
+{
+	std::array<uint8_t, 16> testPayload = { 0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
+		                                    0x08, 0x09, 0x0A, 0x0B, 0x0C, 0x0D, 0x0E, 0x0F };
+	pcpp::RawPacket rawPacket1(testPayload.data(), testPayload.size(), timeval({ 1, 2 }), false);      // 1.000002000
+	pcpp::RawPacket rawPacket2(testPayload.data(), testPayload.size(), timespec({ 1, 1234 }), false);  // 1.000001234
+
+	// Create some pcaps in a nested scope to test cleanup on destruction.
+	{
+		pcpp::PcapFileWriterDevice writerDevDestructor1(EXAMPLE_PCAP_DESTRUCTOR1_PATH, pcpp::LINKTYPE_ETHERNET, false);
+		PTF_ASSERT_EQUAL(writerDevDestructor1.getTimestampPrecision(), pcpp::FileTimestampPrecision::Microseconds,
+		                 enumclass);
+		PTF_ASSERT_TRUE(writerDevDestructor1.open());
+		PTF_ASSERT_EQUAL(writerDevDestructor1.getTimestampPrecision(), pcpp::FileTimestampPrecision::Microseconds,
+		                 enumclass);
+		PTF_ASSERT_TRUE(writerDevDestructor1.writePacket(rawPacket1));
+		PTF_ASSERT_TRUE(writerDevDestructor1.writePacket(rawPacket2));
+
+		pcpp::PcapFileWriterDevice writerDevDestructor2(EXAMPLE_PCAP_DESTRUCTOR2_PATH, pcpp::LINKTYPE_ETHERNET, false);
+		PTF_ASSERT_EQUAL(writerDevDestructor2.getTimestampPrecision(), pcpp::FileTimestampPrecision::Microseconds,
+		                 enumclass);
+		PTF_ASSERT_TRUE(writerDevDestructor2.open());
+		PTF_ASSERT_EQUAL(writerDevDestructor2.getTimestampPrecision(), pcpp::FileTimestampPrecision::Microseconds,
+		                 enumclass);
+		PTF_ASSERT_TRUE(writerDevDestructor2.writePacket(rawPacket1));
+		PTF_ASSERT_TRUE(writerDevDestructor2.writePacket(rawPacket2));
+		writerDevDestructor2.close();
+	}
+
+	// Check that file sizes are equal.  This may fail if the pcpp::PcapFileWriterDevice destructor does not close
+	FILE* writerDevDestructor1 = fopen(EXAMPLE_PCAP_DESTRUCTOR1_PATH, "rb");
+	fseek(writerDevDestructor1, 0, SEEK_END);
+	long pos1 = ftell(writerDevDestructor1);
+	fclose(writerDevDestructor1);
+	FILE* writerDevDestructor2 = fopen(EXAMPLE_PCAP_DESTRUCTOR2_PATH, "rb");
+	fseek(writerDevDestructor2, 0, SEEK_END);
+	long pos2 = ftell(writerDevDestructor2);
+	fclose(writerDevDestructor2);
+	PTF_ASSERT_EQUAL(pos1, pos2);
```

Comment:
Why not use C++11 `ifstream`, `seekg` and `tellg`?

Reply:
Updated to use those.